x/evm/keeper/msg_server.go

```diff
@@ -714,13 +714,14 @@ func (s msgServer) VoteConfirmDeposit(c context.Context, req *types.VoteConfirmD
 	}
 
 	keeper := s.ForChain(chain.Name)
-	pendingDeposit, pollFound := keeper.GetPendingDeposit(ctx, req.PollKey)
+	pendingDeposit, pendingDepositFound := keeper.GetPendingDeposit(ctx, req.PollKey)
 	confirmedDeposit, state, depositFound := keeper.GetDeposit(ctx, common.Hash(req.TxID), common.Address(req.BurnAddress))
+	poll := s.voter.GetPoll(ctx, req.PollKey)
 
 	switch {
```

Comment:
this function is primarily about voting, so the switch should reflect that. Make all cases about the different states the poll can be in. The deposit is modified "incidentally" when the poll state changes, but that should not inform the response to the user